src/adapter/src/error.rs

```diff
@@ -532,6 +535,8 @@ impl AdapterError {
             AdapterError::RtrDropFailure(_) => SqlState::UNDEFINED_OBJECT,
             AdapterError::UnreadableSinkCollection => SqlState::from_code("MZ009"),
             AdapterError::UserSessionsDisallowed => SqlState::from_code("MZ010"),
+            // WIP: What's the correct SqlState for this?
+            AdapterError::ReadOnly => SqlState::READ_ONLY_SQL_TRANSACTION,
```

Comment:
I think this state works, we could say that in read-only mode all transactions are implicitly read-only transactions. Another SQL state that might be applicable is 55000, `object_not_in_prerequisite_state`. For example, PostgreSQL uses this when trying to take certain actions while recovery is in progress: 
- https://github.com/postgres/postgres/blob/3fb59e789dd9f21610101d1ec106ad58095e24f3/src/backend/access/transam/xlogfuncs.c#L298-L302
- https://github.com/postgres/postgres/blob/3fb59e789dd9f21610101d1ec106ad58095e24f3/src/backend/access/transam/xlogfuncs.c#L385-L390

Reply:
I'll go with this and add that as a comment. Also, users are never supposed to see this, so ... 🤷‍♂️

Reply:
Yeah, at the end of the day I don't think it matters that much what error we choose.